### PR TITLE
[WFLY-11250] Add documentation for 'default-ssl-context' attribute.

### PR DESCRIPTION
**Pull request by darranl**
Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:
- [x] Pull Request title is properly formatted: `[WFLY-XYZ] Subject` or `WFLY-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue(s)

https://issues.jboss.org/browse/WFLY-11250

- [x] Pull Request contains description of the issue(s)

Documentation for the new 'default-ssl-context' attribute being added under https://github.com/wildfly/wildfly-core/pull/3570

- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Attached commits represent units of work and are properly formatted

For bigger changes, major and minor component upgrades make sure your PR also meets following requirements:
- [x] Pull Request requires a change to the documentation
- [x] Documentation have been updated accordingly
- [ ] Tests were added to cover changes

Not needed in WildFly, these are in WildFly Core.

For new features ensure as well:
- [x] Analysis was done
- [ ] Test Plan has been done
- [ ] Tests were verified in advance

If you are not an active contributor of the WildFly project you can request sponsorship by one of the members to help guide you through the process.

### CODE REVIEW

File: docs/src/main/asciidoc/_elytron/Using_the_Elytron_Subsystem.adoc [hunk 1]
```diff
@@ -1999,6 +1999,45 @@ use custom implementations of the following components:
...
+javax.net.ssl.SSLContext.getDefault();
+----
+
+By default this `SSLContext` is configured using system properties, however within the WildFly Elytron subsystem it is possible to specify that one of the configured contexts should be associated and used as the default.
+
+To make use of this feature configure your `SSLContext` as normal, the following command can then be used to specify which `SSLContext` should be used as the default.
+
+[source, ruby]
+----
+/subsystem=elytron:write-attribute(name=default-ssl-context, value=client-context)
```
Comment (mchoma) [inline docs/src/main/asciidoc/_elytron/Using_the_Elytron_Subsystem.adoc]: Looking into implementation. Does it mean restart-required will be outputed here? I understand model does not provide this level of configuration. But maybe we should document this mismatch and despite restart-required is printed reload is enough in this case.

Comment (mchoma) [reply docs/src/main/asciidoc/_elytron/Using_the_Elytron_Subsystem.adoc]: I see correctly reload required after writing to attribute
```    "outcome" => "success",
    "response-headers" => {
        "operation-requires-reload" => true,
        "process-state" => "reload-required"
    }
```

Comment (darranl) [reply docs/src/main/asciidoc/_elytron/Using_the_Elytron_Subsystem.adoc]: +1 I didn't add that as we don't tend to all out every reload required situation in the documentation but the restart required was a special case so needed a special mention.

**Review by bstansberry — APPROVED** [review]
I'm going to go ahead and merge this because it LGTM and the conversation above looks resolved. The discussion is about the handling of reload-required vs restart-required, and the PR text seems clear about that, with the text matching what the impl is coded to do and also matching my I believe the consensus in the discussion was.

**Pull request merged**